File: matlab/lib/+Ice/EndpointSelectionType.m
classdef EndpointSelectionType < uint8
    %ENDPOINTSELECTIONTYPE Determines how the Ice runtime sorts proxy endpoints when establishing a connection.
    %
    %   EndpointSelectionType Properties:
    %     Random - The Ice runtime shuffles the endpoints in a random order.
    %     Ordered - The Ice runtime uses the endpoints in the order they appear in the proxy.

    % Copyright (c) ZeroC, Inc.

    enumeration
        %RANDOM The Ice runtime shuffles the endpoints in a random order.
        Random (0)

        %ORDERED The Ice runtime uses the endpoints in the order they appear in the proxy.
        Ordered (1)
    end
    methods (Static)
        function r = ice_getValue(v)
            switch v
                case 0
                    r = Ice.EndpointSelectionType.Random;
                case 1
                    r = Ice.EndpointSelectionType.Ordered;
                otherwise
                    throw(Ice.MarshalException(sprintf('enumerator value %d is out of range', v)));
            end
        end
    end
end

File: java/src/com.zeroc.ice/src/main/java/com/zeroc/Ice/EndpointSelectionType.java
// Copyright (c) ZeroC, Inc.

package com.zeroc.Ice;

/** Determines how the Ice runtime sorts proxy endpoints when establishing a connection. */
public enum EndpointSelectionType {
    /** The Ice runtime shuffles the endpoints in a random order. */
    Random(0),

    /** The Ice runtime uses the endpoints in the order they appear in the proxy. */
    Ordered(1);

    /**
     * Returns the integer value of this enumerator.
     *
     * @return the integer value of this enumerator
     */
    public int value() {
        return _value;
    }

    /**
     * Returns the enumerator corresponding to the given integer value.
     *
     * @param v the integer value of the enumerator
     * @return the enumerator corresponding to the given integer value, or {@code null} if no such enumerator exists
     */
    public static EndpointSelectionType valueOf(int v) {
        switch (v) {
            case 0:
                return Random;
            case 1:
                return Ordered;
        }
        return null;
    }

    private EndpointSelectionType(int v) {
        _value = v;
    }

    private final int _value;
}
